File: Reading/数据结构与算法图解/code/solutions/solution14-1.py
# We've provided the basic Node and LinkedList classes in addition to the actual solution,
# which is the printList function

class Node:

  def __init__(self, data, nextNode=None):
    self.data = data
    self.nextNode = nextNode

class LinkedList:

  def __init__(self, firstNode):
    self.firstNode = firstNode

  def printList(self):
    currentNode = self.firstNode

    while currentNode:
      print(currentNode.data)
      currentNode = currentNode.nextNode

File: Reading/数据结构与算法图解/code/solutions/solution14-1.js
/***
 * Excerpted from "A Common-Sense Guide to Data Structures and Algorithms, Second Edition",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material,
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose.
 * Visit http://www.pragmaticprogrammer.com/titles/jwdsal2 for more book information.
***/
// We've provided the basic Node and LinkedList classes in addition to the actual solution,
// which is the printList function

class Node {

  constructor(data, nextNode=null) {
    this.data = data;
    this.nextNode = nextNode;
  }

}

class LinkedList {
  constructor(firstNode) {
    this.firstNode = firstNode;
  }

  printList() {
    let currentNode = this.firstNode;

    while (currentNode) {
      console.log(currentNode.data);
      currentNode = currentNode.nextNode;
    }
  }
}